File: NirvanaSMS.Api.CSharp/Operations/HeaderOperations.cs
using NirvanaSMS.Api.Core.Base;
using System;
using System.Collections.Generic;
using System.Linq;
using System.Text;
using System.Threading.Tasks;
using NirvanaSMS.Api.Core.Entity;
using NirvanaSMS.Api.Core.Utility;

namespace NirvanaSMS.Api.Operations
{
    public class HeaderOperations : HeaderBase
    {
        public override ProcessResult<string[]> Query(ApiUser apiUser)
        {
            Dictionary<string, string> queryStrings = new Dictionary<string, string>();
            queryStrings.Add("username", apiUser.UserName);
            queryStrings.Add("password", apiUser.ApiPassword);
            HttpRequest request = new HttpRequest(Constants.HeaderQueryUrl, queryStrings, Enums.RequestMethod.Get);
            string response = request.GetResponse();
            ProcessResult<string[]> result = new ProcessResult<string[]>(response);
            if (result.IsSuccess)
            {
                string[] headers = response.Substring(2).Split('|');
                result.Data = headers;
            }
            return result;
        }
    }
}

File: NirvanaSMS.Api.Java/src/main/java/org/nirvanasms/api/operations/HeaderOperations.java
package org.nirvanasms.api.operations;

import java.util.HashMap;
import java.util.Map;

import org.nirvanasms.api.core.base.HeaderBase;
import org.nirvanasms.api.core.entity.ApiUser;
import org.nirvanasms.api.core.utility.Constants;
import org.nirvanasms.api.core.utility.Enums.RequestMethod;
import org.nirvanasms.api.core.utility.HttpRequest;
import org.nirvanasms.api.core.utility.ProcessResult;

public class HeaderOperations extends HeaderBase {
	@Override
	public ProcessResult<String[]> query(ApiUser apiUser)
    {
        Map<String, String> queryStrings = new HashMap<String, String>();
        queryStrings.put("username", apiUser.getUserName());
        queryStrings.put("password", apiUser.getApiPassword());
        HttpRequest request = new HttpRequest(Constants.HeaderQueryUrl, queryStrings, RequestMethod.Get);
        String response = request.getResponse();
        ProcessResult<String[]> result = new ProcessResult<String[]>(response);
        if (result.isSuccess()) {
            String[] headers = response.substring(2).split("\\|");
            result.setData(headers);
        }
        return result;
    }
}
